File: contributions/Legacy/MOO/optimization/ntwOpt/Matlab/mainNewYork.m
%-----------------
% Florian Mueller
% December 2014
%-----------------

ld = LayoutData();
ld = setNewYork(ld);
hd = HydraulicData();
hd = setNewYork(hd,ld);

plotCuts(hd,int64(10),'Plots/','cuts');
plotLayout(ld,'Plots/','layout');
close all;

hs = HydraulicSolution();
fprintf('\nsolve hydraulic mixed integer linear program ...\n\n');
hs = solveHydraulicMixedIntLinProg(ld,hd,hs);

rhd = ReducedHydraulicData();
rhd = setNewYork(rhd,ld,hd,hs);

rhs = ReducedHydraulicSolution();
fprintf('\nsolve reduced hydraulic linear program ...\n\n');
rhs = solveReducedHydraulicLinProg(ld,hd,rhd,rhs);

td = ThermalData();
td = setNewYork(td,ld,hd,rhd,rhs);

ts = ThermalSolution();
fprintf('\nsolve thermal linear program ...\n\n');
ts = solveThermalLinProg(ld,hd,td,ts);

writeToMat(ld,'Mat/ld');
writeToMat(hd,'Mat/hd');
writeToMat(rhd,'Mat/rhd');
writeToMat(td,'Mat/td');
writeToMat(hs,'Mat/hs');
writeToMat(rhs,'Mat/rhs');
writeToMat(ts,'Mat/ts');

File: contributions/Legacy/MOO/optimization/ntwOpt/Python/mainNewYork.py
#-----------------
# Florian Mueller
# December 2014
#-----------------



import LayoutData
import HydraulicData
import ReducedHydraulicData
import ThermalData
import Solution
import subprocess
import os

CodePath = "C:/urben/MOO/"
finalDir = CodePath + "ntwOpt/"
matlabDir = "C:/Program Files/MATLAB/R2014a/bin"
matDir    = finalDir+'/Matlab/Mat/'

ld = LayoutData.LayoutData()
ld.setNewYork()
ld.writeToMat(matDir+'ld')

hd = HydraulicData.HydraulicData()
hd.setNewYork(ld)
hd.writeToMat(matDir+'hd')

if 1:
    print '\nsolve hydraulic mixed integer linear program ...\n'
    mCode = " ".join(["cd('%(finalDir)sMatlab');",
                      "ld = LayoutData();",
                      "ld = setFromMat(ld, '%(matDir)sld');",
                      "hd = HydraulicData();",
                      "hd = setFromMat(hd, '%(matDir)shd');",
                      "hs = HydraulicSolution();",
                      "hs = solveHydraulicMixedIntLinProg(ld, hd, hs);",
                      "hs = writeToMat(hs, '%(matDir)shs');",
                      "exit();"]) % locals()

    subprocess.call([os.path.join(matlabDir, 'matlab.exe'), '-wait', '-nosplash', '-nodesktop', '-r', mCode], shell=True)

hs = Solution.Solution()
hs.setFromMat(matDir+'hs')

rhd = ReducedHydraulicData.ReducedHydraulicData()
rhd.setNewYork(ld,hd,hs)
rhd.writeToMat(matDir+'rhd')

if 1:
    print '\nsolve reduced hydraulic linear program ...\n'
    mCode = " ".join(["cd('%(finalDir)sMatlab');",
                      "ld  = LayoutData();",
                      "ld  = setFromMat(ld, '%(matDir)sld');",
                      "hd  = HydraulicData();",
                      "hd  = setFromMat(hd, '%(matDir)shd');",
                      "rhd = ReducedHydraulicData();",
                      "rhd = setFromMat(rhd, '%(matDir)srhd');",
                      "rhs = ReducedHydraulicSolution();",
                      "rhs = solveReducedHydraulicLinProg(ld,hd,rhd,rhs);",
                      "rhs = writeToMat(rhs, '%(matDir)srhs');",
                      "exit();"]) % locals()
                      
    subprocess.call([os.path.join(matlabDir, 'matlab.exe'), '-wait', '-nosplash', '-nodesktop', '-r', mCode], shell=True)

rhs = Solution.Solution()
rhs.setFromMat(matDir+'rhs')

td = ThermalData.ThermalData()
td.setNewYork(ld,hd,rhd,rhs)
td.writeToMat(matDir+'td')

if 1:
    print '\nsolve thermal linear program ...\n'
    mCode = " ".join(["cd('%(finalDir)sMatlab');",
                      "ld  = LayoutData();",
                      "ld  = setFromMat(ld, '%(matDir)sld');",
                      "hd  = HydraulicData();",
                      "hd  = setFromMat(hd, '%(matDir)shd');",
                      "td = ThermalData();",
                      "td = setFromMat(td, '%(matDir)std\');",
                      "ts = ThermalSolution();",
                      "ts = solveThermalLinProg(ld,hd,td,ts);",
                      "ts = writeToMat(ts, '%(matDir)sts\');",
                      "exit();"]) % locals()
                      
    subprocess.call([os.path.join(matlabDir, 'matlab.exe'), '-wait', '-nosplash', '-nodesktop', '-r', mCode], shell=True)

ts = Solution.Solution()
ts.setFromMat(matDir+'ts')


